uplink/eestream/decode.go

```diff
@@ -192,6 +192,8 @@ func (dr *decodedRanger) Size() int64 {
 
 func (dr *decodedRanger) Range(ctx context.Context, offset, length int64) (_ io.ReadCloser, err error) {
 	defer mon.Task()(&ctx)(&err)
+
+	ctx, cancel := context.WithCancel(ctx)
```

Comment:
Maybe add a comment on line 204 saying what concrete type the 'rr' is so that the context cancellation stuff makes more sense.